### ISSUE TITLE
Support for estimation of expected survival time and similar measures

### ISSUE THREAD

**Issue by adibender** [adibender/pammtools#136]
see also: https://journal.r-project.org/archive/2019/RJ-2019-042/index.html

**Comment by fabian-s** [comment]
> see also: https://journal.r-project.org/archive/2019/RJ-2019-042/index.html
o wow.... these trivial tiny weird things survival people get published.... ;) 

ok, but, (numerically) integrating S(t) as mentioned in the comments to PR #131 would seem the way to go since this is fairly easy to do based on existing functionality in pammtools? that other weird stuff (put a GAM on top of our rankings and times etc) only works for PH models anyway, right? 
these `coxed` people simply integrate to max(event times, censoring times). That seems simplistic/dangerous to me.

**Comment by adibender** [comment]
Yeah, i was thinking that their stuff is not necessary for us, but wanted to link for reference.  

> these coxed people simply integrate to max(event times, censoring times). That seems simplistic/dangerous to me

true, but we also only have S(t) estimates until the last interval (which using default cuts will end at the last max(event times, censoring times). 

Also, once we have the estimates for E(t|x) we also need an estimate for SE (potentially for all 3 SE types) ...

**Comment by fabian-s** [comment]
> true, but we also only have S(t) estimates until the last interval (which using default cuts will end at max(event times, censoring times).

true, too, all i'm saying it really does not make sense to set the upper limit of the integration to a Tmax for which S(Tmax) is not already almost zero, or you obviously get a large downward bias.  

> Also, once we have the estimates for E(t|x) we also need an estimate for SE (potentially for all 3 SE types) ...

seems exceedingly easy for `sim`...? for `direct`, we'd just integrate the upper/lower pointwise CI of S(t)? for `delta`, we need to run Maple I think :face_with_head_bandage:

**Comment by adibender** [comment]
maybe of interest: https://onlinelibrary.wiley.com/doi/abs/10.1002/%28SICI%291097-0258%2819960130%2915%3A2%3C171%3A%3AAID-SIM146%3E3.0.CO%3B2-U
